servant-server/servant-server.cabal

```diff
@@ -164,3 +164,5 @@ test-suite doctests
  if impl(ghc >= 8.2)
    x-doctest-options: -fdiagnostics-color=never
  include-dirs: include
+ other-modules:
```

Comment:
don't do this. this breaks `sdist. I fixed the CI already. Will make it a bit cleaner soon-ish.

Reply:
OK, though if we need to use `Wno-missing-home-modules`, can we put that in the cabal or stack file rather than just `travis.yaml`? Otherwise everyone is going to run into compilation errors/warnings locally.

(Also, I'd love to understand what's going on better!)

Reply:
Yes, I'll do that (putting warning suppression in a proper place) PR separately.

- Cabal 2.0 passes `-Wmissing-home-modules`, because it's good idea.
- So `Build_foo`, or `Paths_exe` or `Lexer` are reported missing.
- So you add them as `other-modules`
- But it breaks `cabal sdist`, so you declare `autogen-modules` (feature in `Cabal-2.0`)
   - this is *ok-ish*, because old-Cabal's would just omit the field, and you don't use them for `sdist`
- Except when `build-type: Custom` the situation is trick-ier. Then `sdist` command uses system `Cabal`, and not the one `cabal-install` or `stack` is compiled with.
    - Yet, we don't do `sdist` check in CI, so it's not a problem now
    - but I think we should, because that will catch forgotten test files etc.
    - `-Wmissing-home-modules` is not enough by itself, if we edit `extra-source-files` for example

There is no elegant solution atm. Proper ones are:
- `cabal new-doctest` (and `stack` support as a follow-up)
- ability to configure `ghc-options` per component

Reply:
:+1: on making `sdist` part of CI. From the sound of it `-Wno-missing-home-modules` is the way to go for now.